pkgs/development/libraries/arrow-cpp/default.nix

```diff
@@ -228,14 +237,35 @@ stdenv.mkDerivation rec {
     in
     lib.optionalString doInstallCheck "-${builtins.concatStringsSep ":" filteredTests}";
   installCheckInputs = [ perl which sqlite ] ++ lib.optional enableS3 minio;
+
+  postBuild = lib.optionalString (doInstallCheck && enableCuda) ''
+    for file in $(find . -name '*-test' -o -name '*-tests' -type f -executable); do
+      addOpenGLRunpath "$file"
+    done
+  '';
+
   installCheckPhase =
     let
       excludedTests = lib.optionals stdenv.isDarwin [
         # Some plasma tests need to be patched to use a shorter AF_UNIX socket
         # path on Darwin. See https://github.com/NixOS/nix/pull/1085
         "plasma-external-store-tests"
         "plasma-client-tests"
-      ] ++ [ "arrow-gcsfs-test" ];
+      ] ++ [
+        "arrow-gcsfs-test"
+      ] ++ lib.optionals enableCuda [
+        # we can run this but it requires access to /run/opengl-driver and some
+        # other things under /dev, see https://github.com/NixOS/nix/issues/4489
+        "arrow-cuda-test"
```

Comment:
This is the sort of thing that I had in mind for https://github.com/samuela/cuda-nix-testsuite/, but it's currently beta-status and adding tests to it is entirely optional.